predicates/primitive/shutdown.go

```diff
@@ -0,0 +1,51 @@
+package primitive
+
+import (
+	"net/http"
+	"os"
+	"os/signal"
+	"sync/atomic"
+	"syscall"
+
+	"github.com/zalando/skipper/predicates"
+	"github.com/zalando/skipper/routing"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type shutdown struct {
+	inShutdown int32
+}
+
+// NewShutdown provides a predicate spec to create predicates
+// that evaluate to true if Skipper is shutting down
+func NewShutdown() routing.PredicateSpec {
+	s, _ := newShutdown()
+	return s
+}
+
+func newShutdown() (routing.PredicateSpec, chan os.Signal) {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+	s := &shutdown{}
+	go func() {
+		<-sigs
+		log.Infof("Got shutdown signal for %s predicates", s.Name())
+		atomic.StoreInt32(&s.inShutdown, 1)
```

Comment:
I think for an int64 on 64bit platforms you do not need an atomic or guards, because it's an aligned access and there is only one read and one write operation that should be in an aligned access. From that pov you should also try to omit 32bit access, because it's slower than 64bit access (32bit AFAIK needs an additional shift operation).
I also don't see a problem if we would have here a race, that would create healthcheck true, but you already got a shutdown signal in the same frame of a couple of µs.

Reply:
http.Server uses int32 for this purpose. On x86 atmic.Store/Load are inlined into three instructions https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/runtime/internal/atomic/atomic_amd64.s;l=159-163  

> I also don't see a problem if we would have here a race

I just want clean code, the difference would be nanoseconds

Reply:
Also https://golang.org/ref/mem#tmp_1

Reply:
:+1: